backend/services/GraphService.ts

```diff
@@ -1,174 +1,118 @@
+      .map((userGraph) => userGraph.dsnpGraphEdges?.map((edge) => edge.userId.toString()))
+      .filter((item): item is string[] => item !== undefined)
+      .flat();
+    logger.debug({ followList }, 'GraphService: getPublicFollows: processed followList');
     return followList;
   }
 
-  public async follow(actorId: string, objectId: number): Promise<void> {
-    console.log('Follow Request', { actorId, objectId });
-    const api = await getApi();
-    const schemaId = getSchemaId(AnnouncementType.PublicFollows);
-    const resp = await api.rpc.statefulStorage.getPaginatedStorage(actorId, schemaId);
-
-    const pages = resp.map((page) => page.toJSON());
-
-    const followPages = pages.map((page) => {
-      try {
-        return this.inflatePage(page.payload).map((x: GraphEdge) => x.userId);
-      } catch (e) {
-        console.error('Failed to parse public follows...', e);
-        return [];
-      }
-    });
-
-    for (const page of followPages) {
-      if (page.includes(objectId)) {
-        return;
-      }
-    }
-
-    let pageNumber = pages.length - 1;
-
-    let upsertEdges: GraphEdge[] = [];
-    let hash = null;
-
-    // Check if we should use a new page
-    if (pageNumber === -1 || followPages[pageNumber].length >= 93) {
-      pageNumber = pageNumber >= 0 ? pageNumber + 1 : 0;
-    } else {
-      const lastPage = pages[pageNumber];
-      upsertEdges = this.inflatePage(lastPage.payload);
-      hash = lastPage.content_hash;
-    }
-
-    upsertEdges.push({
-      userId: objectId,
-      since: Math.floor(Date.now() / 1000),
-    });
-    console.log('upsertEdges', upsertEdges);
-
-    const encodedPage = this.deflatePage(upsertEdges);
-    const payload = '0x' + encodedPage.toString('hex');
-
-    const tx = api.tx.statefulStorage.upsertPage(actorId, schemaId, pageNumber, hash, payload);
-    // Do NOT wait for all the callbacks. Assume for now that it will work...
-    await api.tx.frequencyTxPayment
-      .payWithCapacity(tx)
-      .signAndSend(getProviderKey(), { nonce: await getNonce() }, ({ status, dispatchError }) => {
-        if (dispatchError) {
-          console.error('Graph ERROR: ', dispatchError.toHuman());
-        } else if (status.isInBlock || status.isFinalized) {
-          console.log('Graph Updated: ', status.toHuman());
-        }
-      });
+  /**
+   * Posts a follow request from an actor to an object.
+   *
+   * @param actorId - The ID of the actor.
+   * @param objectId - The ID of the object.
+   * @returns A Promise that resolves to void.
+   */
+  public async postFollow(actorId: string, objectId: number): Promise<void> {
+    logger.debug({ actorId, objectId }, 'Follow Request');
+    const providerGraphDto: ProviderGraphDto = {
+      dsnpId: actorId,
+      connections: {
+        data: [
+          {
+            dsnpId: objectId.toString(),
+            privacyType: 'public',
+            direction: 'connectionTo',
+            connectionType: 'follow',
```

Comment:
Is there a reason that we don't want to have these strings be enums or held somewhere more global to be reused?

Reply:
Really, these should be defined by `graph-service` and exported/imported via the OpenAPI document. I'll look into it; we do something similar with `content-watcher`.